### PR TITLE
map_overlap: Don't rechunk axes without overlap

### PR DESCRIPTION
**Pull request by dcherian**
- [ ] Closes #xxxx
- [x] Tests added / passed
- [x] Passes `black dask` / `flake8 dask`

### CODE REVIEW

File: dask/array/overlap.py [hunk 1]
```diff
@@ -817,9 +817,7 @@ def coerce_depth(ndim, depth):
     if isinstance(depth, tuple):
         depth = dict(zip(range(ndim), depth))
     if isinstance(depth, dict):
-        for i in range(ndim):
-            if i not in depth:
-                depth[i] = 0
+        depth = {ax: depth.get(ax, default) for ax in range(ndim)}
```
Comment (dcherian) [inline dask/array/overlap.py:820]: The bug is here
https://github.com/dask/dask/blob/1c8b5eafe2be16961472e34fff0c78942f0c6647/dask/array/overlap.py#L527-L531

Iterating over `depth2.values` assumes that `depth2` has keys, values in axes order which it did not.

File: dask/array/overlap.py [hunk 2]
```diff
@@ -841,7 +839,5 @@ def coerce_boundary(ndim, boundary):
     if isinstance(boundary, tuple):
         boundary = dict(zip(range(ndim), boundary))
     if isinstance(boundary, dict):
-        for i in range(ndim):
-            if i not in boundary:
-                boundary[i] = default
+        boundary = {ax: boundary.get(ax, default) for ax in range(ndim)}
```
Comment (dcherian) [inline dask/array/overlap.py:842]: not sure there is an issue with boundary but made the "fix" in any case

**Review by jsignell — APPROVED** [review]
This looks good to me. Is there an issue that should be linked. Not a problem if not, but just checking.

**Comment by dcherian** [comment]
No issue that I could find. The bug causes inefficient-but-correct computations or complete failure if you provided `chunks` to `map_overlap` that didn't match whatever `overlap` ended up rechunking to.

**Pull request merged**